### DISCUSSION TITLE
Is it a good idea to get rid of CONFIG?

### DISCUSSION THREAD

**Discussion by Varniex** [3b1b/manim#1392] [General]
Hi,

I was wondering about getting rid of `CONFIG` process in manimgl just like ManimCommunity did. But, I'm not sure if it's a good idea or not.

**Comment by TonyCrane** [comment]
In my opinion, this may not be a good idea.
Although using something like `CONFIG` does not conform to the normal python package specification, it is very convenient in use.
Manim's class initialization requires many attributes, and there are also many inherited attributes between classes. Using such a `CONFIG` dictionary for inheritance is a wonderful way for me.
If everything needs to be written in `__init__`, it seems to me a bit complicated. Moreover, when using `CONFIG`, it also supports converting the incoming `kwargs` into attributes.

**Reply by Varniex** [reply]
Initially, it seemed a little unpythonic way. But, yeah, as you said, it is convenient and helpful for the subclasses to make attributes.
